controller/rpc/scanner.go

```diff
@@ -15,90 +15,170 @@ import (
 )
 
 type scannerAct struct {
-	scanner  *share.CLUSScanner
-	scanning uint32
+	scanner            *share.CLUSScanner
+	activeScannerTasks int // Indicate how much scanner task is currently run
 }
 
-var scanners map[string]*scannerAct = make(map[string]*scannerAct)
-var scanMutex sync.RWMutex
+type ScannerManager struct {
+	activeScanners map[string]*scannerAct
+	maxConns       int
+	mutex          sync.RWMutex
+	availableCh    chan struct{}
```

Comment:
minor: I think your design is basically a credit system.  When a new scanner is added, it gives you credits to run scan.  When a scanner is removed, those credits are removed again.  Each scan job requires 1 credit in order to run.  Not sure if it's a good idea to give it a better naming.